pkg/pipelineascode/match.go

```diff
@@ -57,7 +57,7 @@ func (p *PacRun) matchRepoPR(ctx context.Context) ([]matcher.Match, *v1alpha1.Re
 	// allow webhook providers users to have a global webhook secret to be used,
 	// so instead of having to specify their in Repo each time, they use a
 	// shared one from pac.
-	if repo.Spec.GitProvider != nil {
+	if repo.Spec.GitProvider != nil && p.event.InstallationID == 0 {
```

Comment:
in case of github webhook it would be `-1`
 https://github.com/openshift-pipelines/pipelines-as-code/blob/ab509087dd251e4b857bc65f76b8e25a0a9f6b2d/pkg/provider/github/events.go#L117

Reply:
not sure why i made it return -1 😅 if installation id not found
https://github.com/openshift-pipelines/pipelines-as-code/blob/ab509087dd251e4b857bc65f76b8e25a0a9f6b2d/pkg/provider/github/events.go#L94

Reply:
I am not sure either :D but 0 is sometime mean undefined/None in some languages so -1 is not bad...